python/neuroglancer/pipeline/threaded_queue.py

```diff
@@ -129,7 +130,11 @@ def _consume_queue(self, terminate_evt):
         continue # periodically check if the thread is supposed to die
 
       fn = partial(fn, interface)
-      self._consume_queue_execution(fn)
+
+      try:
+        self._consume_queue_execution(fn)
+      except Exception as err:
```

Comment:
Subclasses have the opportunity to handle the exception by overriding `_consume_queue_execution` as in `TaskQueue`. I don't reraise here to avoid by chance killing all the threads and `queue.join` never unblocking on the main thread.